drafts/0007-RFC-BAG-OPERATORS.md

```diff
@@ -0,0 +1,564 @@
+- Feature Name: Bag Operators: UNION, INTERSECT, EXCEPT
+- Start Date: 2022-06-07
+
+# Summary
+[summary]: #summary
+
+This RFC defines the PartiQL specification for the operators: UNION, INTERSECT, and EXCEPT.
+
+# Motivation
+[motivation]: #motivation
+
+The operators UNION, INTERSECT, and EXCEPT have been defined since SQL-92 specifications, yet these are missing from the PartiQL spec.
+In SQL, these operators are the typical bag (set if DISTINCT) union, intersect, and difference respectively for *union-compatible* relations. Unlike SQL, PartiQL is schemaless and therefore these operators are valid for arbitrary expressions — including a SFW block.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Each bag operator has the form *q S q'* where *q* and *q'* are of type *\<query\>* and *S* is the operator. Additionally, the operator may be suffixed with *ALL* which indicates the output may have duplicate elements. In its absence, duplicates are eliminated from the final result. Arguments of the operators are coerced into bags with the following behavior.  
+
+- *NULL* and *MISSING* become the empty bag *<< >>*
+- A scalar value *v* becomes the singleton bag *<< v >>*
+- An array is coerced to a bag by discarding ordering
+
+The behavior of bag operators is specified by permissive or conventional type checking mode. The typing mode determines if heterogenous arguments are allowed or not allowed. Conventional typing mode will produce an error if a bag operator is used on heterogenous types (??). All six operators are defined in Figure 1.
```

Comment:
Why would, even in conventional (strict) mode would heterogenous types not work?  Do you just mean at the container level?